qos/noop/noop.go

```diff
@@ -36,6 +36,15 @@ func (_ NoOpQoS) ParseHTTPRequest(_ context.Context, httpRequest *http.Request)
 	}, true
 }
 
+// ParseWebsocketRequest builds a request context from the provided WebSocket request.
+// This method implements the gateway.QoSService interface.
+// TODO_HACK(@commoddity, WebSockets): Utilize this method once the Shannon protocol supports websocket connections.
```

Comment:
Maybe we should jsut return an error for noop?

Reply:
I don't think we should. The reason being that there may be WebSocket enabled services without an existing QoS implementation and should still allow them to function in that case, albeit with random endpoint selection. Returning an error would effectively make them unusable until we build a QoS implementation for all of them.